pages/shared/studentGroupAssessmentHelpers/studentGroupAssessmentHelpers.js

```diff
@@ -0,0 +1,137 @@
+var ERR = require('async-stacktrace');
+
+var sqldb = require('../../../prairielib/lib/sql-db');
+var sqlLoader = require('../../../prairielib/lib/sql-loader');
+var sql = sqlLoader.loadSqlEquiv(__filename);
+
+module.exports.getConfigInfo = function (req, res, next) {
+  var params = {
+    assessment_id: res.locals.assessment.id,
+    user_id: res.locals.user.user_id,
+  };
+
+  sqldb.query(sql.get_config_info, params, function (err, result) {
+    if (ERR(err, next)) return;
+    res.locals.permissions = result.rows[0];
+    res.locals.minsize = result.rows[0].minimum || 0;
+    res.locals.maxsize = result.rows[0].maximum || 999;
+    sqldb.query(sql.get_group_info, params, function (err, result) {
+      if (ERR(err, next)) return;
+      res.locals.groupsize = result.rowCount;
+      res.locals.needsize = res.locals.minsize - res.locals.groupsize;
+      if (res.locals.groupsize > 0) {
+        res.locals.group_info = result.rows;
+        res.locals.join_code =
+          res.locals.group_info[0].name + '-' + res.locals.group_info[0].join_code;
+        res.locals.start = false;
+        if (res.locals.needsize <= 0) {
+          res.locals.start = true;
+        }
+      }
+      if (res.locals.groupsize >= res.locals.maxsize) {
+        res.locals.used_join_code = true;
+      } else {
+        res.locals.used_join_code = undefined;
+      }
+      next();
+    });
+  });
+};
+
+module.exports.joinGroup = function (req, res, next) {
```

Comment:
As above, you should rename `next` to `callback`.

Reply:
Part of the callback fixes in [8652798](https://github.com/PrairieLearn/PrairieLearn/pull/5290/commits/86527983f782382ca918fb20f3c9b995e3862239)